### ISSUE TITLE
Git was not found, check that it is installed and in your PATH env.

### ISSUE THREAD

**Issue by LZY7977** [composer/composer#11448]
Hi composer team,

I encountered following error when deploying composer app with custom git repository to latest PHP 8.1 platform on AWS Elastic Beanstalk. 
```
In Git.php line 484:

  Failed to clone https://git.drupalcode.org/sandbox/#{some_repo}.git, gi
  t was not found, check that it is installed and in your PATH env.

  sh: git: command not found
```
The composer version is `2.3.5` based on the [release note](https://docs.aws.amazon.com/elasticbeanstalk/latest/platforms/platform-history-php.html). I logged on to the EC2 instance and was able to verify that git was indeed installed and git is in the `PATH` environment variable

```
sh-4.2$ git --version
git version 2.39.2
sh-4.2$ which git
/usr/bin/git
sh-4.2$ echo $PATH
/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin
```

I installed various versions of composer on AWS EB and re-deployed source code. The `git` issue could be replicated from composer v2.2.0 and above. Everything works fine from v2.1.14 and below, which made me wonder if composer team has made any changes about `$PATH` in v2.2.0.

- Starting from v2.2.0, do we need to explicitly set `$PATH` when installing dependencies through composer?
- How does composer read the `$PATH` variable? Would the value for `$PATH` be different if `composer install` is executed differently, say directly through bash shell versus through golang?
- Looking at composer-2.2.0 release log, I thought this [change](https://github.com/composer/composer/pull/10218) might be related. However, I am not able to confirm this, and it would be good if someone from composer team could confirm this. 

Any input or suggestion would be appreciated, Thanks!

**Comment by Seldaek** [comment]
This exception is thrown like this:

        if (0 !== $this->process->execute('git --version', $ignoredOutput)) {
            throw new \RuntimeException(Url::sanitize('Failed to clone ' . $url . ', git was not found, check that it is installed and in your PATH env.' . "\n\n" . $this->process->getErrorOutput()));
        }

So you can see that we just execute `git --version` and check for an error, and then the error output is appended which in this case is `sh: git: command not found` coming from your shell.

I'm not entirely sure what might lead to this issue.. but maybe try with the latest Composer too?

> How does composer read the $PATH variable? Would the value for $PATH be different if composer install is executed differently, say directly through bash shell versus through golang?

It receives the PATH env var from whatever process executes it, so perhaps in golang you're executing it without env vars present and then PATH is not available to Composer?

**Comment by LZY7977** [comment]
I tried composer-v2.2.0, v2.3.5 and v2.5.5, they all failed with the `git not found` issue. I also tried composer-v2.1.14 and v2.0.13, they succeeded without the git issue. Therefore, I am thinking there might be some changes from composer side from v2.2.0.

> It receives the PATH env var from whatever process executes it, so perhaps in golang you're executing it without env vars present and then PATH is not available to Composer?

If composer is executed through golang and we are not directly setting `$PATH` in golang, would composer still be able to read `$PATH`?
